### DISCUSSION TITLE
Feature Request: Support objects that implement dataframe interchange protocol in replacement scan

### DISCUSSION THREAD

**Discussion by mariosasko** [duckdb/duckdb#7786] [Ideas]
This would make extending the replacement scan mechanism to work with data frame libraries other than Pandas (and PyArrow/Polars) simple.

Info on the dataframe interchange protocol:
* [RFC](https://data-apis.org/blog/dataframe_protocol_rfc/)
* [Official docs](https://data-apis.org/dataframe-protocol/latest/index.html)

**Comment by Tishj** [comment]
I read something on this a while back, I think this was spearheaded by pandas + polars, no?
I think it's a great idea, we tried something like this a while back by adding support for Protocols in python, to make it possible to support duck-typed pandas dataframes and the like, but that failed because the candidates were lacking methods that we required.

Since this is a fleshed out protocol, this sounds a lot more promising

**Comment by mariosasko** [comment]
> I think this was spearheaded by pandas + polars, no?

And Arrow, I think.

And I'm listing some more projects that support it: Modin, Vaex, Altair, cuDF, HF Datasets (soon), etc.

**Comment by mariosasko** [comment]
@Tishj This should be much easier to support now thanks to a new PyArrow [C data interface](https://arrow.apache.org/docs/dev/format/CDataInterface/PyCapsuleInterface.html) (added in version 14.0.0), which arbitrary Python objects can implement.
